### ISSUE TITLE
Switching to a past change where a file has not been ignored adds it to the change.

### ISSUE THREAD

**Issue by peacememories** [jj-vcs/jj#3204]
## Description

When new files are added to a gitignore, these become kind of a boobytrap for editing old changes. As soon as you switch to an old change where the file are not ignored yet, they are added to the change since they are in the working copy. This means that, without actively changing anything, and mostly without feedback, you have added files to the repo which you never wanted to be in there.

This is especially awkward when automating commands over a set of changes (I did that to automate `pre-commit` runs for the branch I am working on).

One possible solution to this: When switching between changes, find all files which have been previously `.gitignore`d, and delete them. I realize that this might trash caches for people on a regular basis, but I think this is preferable to accidentally checking in files without noticing.

## Steps to Reproduce the Problem

1. Find a change where a specific file does not exist, let's call it `c`
1. `jj new c`
1. `echo "newfile" >> .gitignore`
2. `touch newfile`
3. `jj edit c`

## Expected Behavior

`newfile` should not be automatically added to the commit

## Actual Behavior

`newfile` is added to the commit.

## Specifications

- Platform: `NixOS 23.11`
- Version: `jj 0.13.0`

**Comment by martinvonz** [comment]
@yuja, do you think we should reopen this bug and close #323? That bug has a very long discussion and I can't remember all of it, but I *think* it's all done except for the solution to the problem described in this bug.

**Comment by yuja** [comment]
> do you think we should reopen this bug and close [#323](https://github.com/jj-vcs/jj/issues/323)? That bug has a very long discussion and I can't remember all of it, but I _think_ it's all done except for the solution to the problem described in this bug.

You mean #323 is resolved by `snapshot.auto-track`? That makes sense to me.

**Comment by martinvonz** [comment]
> You mean [#323](https://github.com/jj-vcs/jj/issues/323) is resolved by `snapshot.auto-track`? That makes sense to me.

Yes, in combination with #5138 (showing untracked files in `jj status`). Alright, I'll reopen this and close the other one then.
